aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -209,7 +209,19 @@ void check_floating_or_complex_dtype(const char* name, ScalarType dtype) {
 
 TORCH_META_FUNC2(mean, dim)
 (const Tensor& self, IntArrayRef dim, bool keepdim, optional<ScalarType> opt_dtype) {
-  check_floating_or_complex_dtype("mean", self.scalar_type());
+  auto in_dtype = at::native::get_dtype_from_self(self, opt_dtype, true);
+
+  std::string opt_dtype_str("None");
```

Comment:
Probably want to hide this computation in a conditional that checks the error conditions so we're not creating this string on every call